transport/src/main/java/io/netty/channel/DefaultChannelConfig.java

```diff
@@ -68,6 +69,8 @@
     private volatile boolean autoClose = true;
     private volatile int writeBufferHighWaterMark = 64 * 1024;
     private volatile int writeBufferLowWaterMark = 32 * 1024;
+    private volatile WriteBufferWaterMark writeBufferWaterMark =
```

Comment:
I think you should remove the local members and just have package private set methods.